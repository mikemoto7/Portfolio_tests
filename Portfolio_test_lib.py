
from test_flags import _product_called_from_test
_product_called_from_test = True


import sys
sys.path.append("../lib_for_test")

from general_test_lib import do_test_runstring
import general_test_lib

general_test_lib.product_dir_path = "../Portfolio"







