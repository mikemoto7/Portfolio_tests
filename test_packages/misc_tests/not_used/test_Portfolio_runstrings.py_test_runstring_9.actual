
Processing: user_runstring = ./Portfolio.py --price_web_update --inv diff --pfile demo_data_file.xml

Processing: --price_web_update
NOTE: Detected in columnize_output(), row 2 is missing a column: 11 instead of 12.  Row: ['---Energy---', '', '', '', '', '', '', '', '', '', ''].  Will pad the row and continue.

Processing: --price_web_update --inv diff demo_data_file.xml 
---------------------------------------------------------------------------------------------------------------------------
Asset                           Symb  Own  iID  Account_ID  Ret_Class  Curr_Tot1  Curr_Tot2   ShPr1   ShPr2     Diff  Diff%
---Energy---                                                                                                               
Vanguard Energy                VGELX  os1   20       15777    Non_Ret   19898.00   19070.00   99.49   95.35   -828.0  -4.16
Vanguard Energy                VGENX  os1   42       35777   Trad_IRA   21208.00   20328.00   53.02   50.82   -880.0  -4.15
Vanguard Energy                VGENX  os1   52       45777   Roth_IRA   26510.00   25410.00   53.02   50.82  -1100.0  -4.15
Vanguard Energy                VGENX  os2   59       52777   Roth_IRA   30221.40   28967.40   53.02   50.82  -1254.0  -4.15
---Global---                                                                                                               
Fidelity Canada                FICDX  os2   39       34777   Trad_IRA   19995.30   19250.40   51.27   49.36   -744.9  -3.73
Vanguard Global Equity         VHGEX  os2    5        4777    SEP_IRA    2348.10    2526.30   26.09   28.07    178.2   7.59
Vanguard Global Equity         VHGEX  os2   23       18777    Non_Ret    6000.70    6456.10   26.09   28.07    455.4   7.59
Vanguard Global Equity         VHGEX  os2   47       40777   Trad_IRA   11740.50   12631.50   26.09   28.07    891.0   7.59
Vanguard Global Equity         VHGEX  os1   58       51777   Trad_IRA   14610.40   15719.20   26.09   28.07   1108.8   7.59
Vanguard Global Min Volatile   VMVFX  os1   24       19777    Non_Ret    2937.60    3081.60   12.24   12.84    144.0    4.9
---Healthcare---                                                                                                           
Vanguard Healthcare            VGHAX  os2    1         777    SEP_IRA    4135.50    4312.00   82.71   86.24    176.5   4.27
Vanguard Healthcare            VGHAX  os1   26       21777    Non_Ret   21504.60   22422.40   82.71   86.24    917.8   4.27
---Individual_Stocks---                                                                                                    
Netflix                         NFLX  os1   18       13777    Non_Ret   25878.60   28693.80  143.77  159.41   2815.2  10.88
---Large_Blend---                                                                                                          
Vanguard Dividend Growth       VDIGX  os2   45       38777   Trad_IRA   10367.30   10745.70   24.11   24.99    378.4   3.65
Vanguard Dividend Growth       VDIGX  os2   55       48777   Trad_IRA   12778.30   13244.70   24.11   24.99    466.4   3.65
Vanguard Total Stock Mkt Idx   VTSAX  os2   35       30777    Non_Ret   20363.00   21021.00   58.18   60.06    658.0   3.23
---Large_Growth---                                                                                                         
Fidelity Contrafund            FCNTX  os1   13        9777    Non_Ret   14644.00   15867.60   104.6  113.34   1223.6   8.36
Fidelity Contrafund            FCNTX  os1   36       31777   Trad_IRA   37656.00   40802.40   104.6  113.34   3146.4   8.36
Fidelity Contrafund            FCNTX  os1   38       33777  Trad_401K   39748.00   43069.20   104.6  113.34   3321.2   8.36
Vanguard Growth Idx            VIGAX  os2   25       20777    Non_Ret   15197.50   16265.00   60.79   65.06   1067.5   7.02
---Large_Value---                                                                                                          
Vanguard Equity Income         VEIRX  os2   21       16777    Non_Ret   14628.60   15023.40   69.66   71.54    394.8    2.7
Vanguard Equity Income         VEIPX  os1   46       39777   Trad_IRA   14621.20   15017.20   33.23   34.13    396.0   2.71
Vanguard Equity Income         VEIPX  os2   57       50777   Trad_IRA   18276.50   18771.50   33.23   34.13    495.0   2.71
---Mid_Blend_Aggress---                                                                                                    
Vanguard Extended Mkt Idx      VEXAX  os1   22       17777    Non_Ret   16662.80   16816.80   75.74   76.44    154.0   0.92
Vanguard Mid-Cap Idx           VIMAX  os2   27       22777    Non_Ret   46213.20   47247.30  171.16  174.99   1034.1   2.24
Vanguard Strategic Equity      VSEQX  os2   33       28777    Non_Ret   11097.90   11035.20   33.63   33.44    -62.7  -0.56
---Mid_Growth---                                                                                                           
Fidelity New Millennium        FMILX  os2   14       10777    Non_Ret    5613.00    5776.50   37.42   38.51    163.5   2.91
Vanguard Capital Opportunity   VHCOX  os2   19       14777    Non_Ret   10831.90   11580.50   57.01   60.95    748.6   6.91
Vanguard Capital Opportunity   VHCAX  os1   50       43777   Rollover   63168.00   67540.80   131.6  140.71   4372.8   6.92
Vanguard Mid-Cap Growth        VMGRX  os1    8        7777   Trad_IRA    2817.60    2929.20   23.48   24.41    111.6   3.96
Vanguard Mid-Cap Growth        VMGRX  os2    9        8777   Trad_IRA    3052.40    3173.30   23.48   24.41    120.9   3.96
---Mid_Value---                                                                                                            
Fidelity Low-Priced Stock      FLPSX  os2   37       32777   Trad_IRA   18825.60   19484.20   50.88   52.66    658.6    3.5
Vanguard Capital Value         VCVLX  os1   44       37777   Trad_IRA    5161.80    5216.40   12.29   12.42     54.6   1.06
Vanguard Capital Value         VCVLX  os1   54       47777   Trad_IRA    6390.80    6458.40   12.29   12.42     67.6   1.06
Vanguard Mid-Cap Value         VMVAX  os1   28       23777    Non_Ret   14677.60   14798.00   52.42   52.85    120.4   0.82
Vanguard Selected Value        VASVX  os1   30       25777    Non_Ret    9027.00    9180.00   30.09    30.6    153.0   1.69
---REIT---                                                                                                                 
Vanguard REIT                  VGSLX  os1    4        3777    SEP_IRA    9502.40    9178.40  118.78  114.73   -324.0  -3.41
Vanguard REIT                  VGSIX  os2    7        6777   Trad_IRA    3062.40    2957.90   27.84   26.89   -104.5  -3.41
Vanguard REIT                  VGSLX  os2   29       24777    Non_Ret   34446.20   33271.70  118.78  114.73  -1174.5  -3.41
Vanguard REIT                  VGSIX  os2   43       36777   Trad_IRA   11414.40   11024.90   27.84   26.89   -389.5  -3.41
Vanguard REIT                  VGSLX  os2   49       42777   Trad_IRA   55826.60   53923.10  118.78  114.73  -1903.5  -3.41
---Small_Blend---                                                                                                          
Vanguard Small Idx             NAESX  os2    3        2777   Trad_IRA    4477.20    4493.30   63.96   64.19     16.1   0.36
Vanguard Tax-Managed Sml Val   VTMSX  os1   34       29777    Non_Ret   19019.60   18907.40   55.94   55.61   -112.2  -0.59
---Small_Growth---                                                                                                         
Vanguard Small Growth          VSGAX  os2   51       44777   Rollover   24093.30   24794.00   49.17    50.6    700.7   2.91
Vanguard Small Growth          VSGAX  os2   53       46777   Roth_IRA   25076.70   25806.00   49.17    50.6    729.3   2.91
Vanguard Small Growth Idx      VSGAX  os2   31       26777    Non_Ret   15242.70   15686.00   49.17    50.6    443.3   2.91
Vanguard Small Growth Idx      VISGX  os1   48       41777   Trad_IRA   18091.80   18616.20   39.33   40.47    524.4    2.9
---Small_Value---                                                                                                          
Vanguard Small-Cap Val Idx     VISVX  os1    6        5777   Trad_IRA    2970.00    2920.00    29.7    29.2    -50.0  -1.68
Vanguard Small-Cap Val Idx     VSIAX  os1   32       27777    Non_Ret   17036.80   16748.80   53.24   52.34   -288.0  -1.69
---Stats---                                                                                                                
Totals                            49                                    859068.8   878260.7                  19191.9       
Totals diff percent                                                                                                    2.23
Avg diff percent                                                                                                       2.27
Count pos diff percent                                                                                                   35
Count neg diff percent                                                                                                   14
Count neutral diff percent                                                                                                0

