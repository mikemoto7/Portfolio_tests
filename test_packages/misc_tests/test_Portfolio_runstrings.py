import pytest

from Portfolio_test_lib import do_test_runstring


# @pytest.mark.usefixtures("oneTimeSetUp", "setUp")
@pytest.mark.usefixtures()
class TestPortfolio():

    # @pytest.fixture(autouse=True)
    # def classSetup(self, setup):
    #     pass



    # Testcases interactive:

    '''
    def test_runstring_cl(self):
        do_test_runstring("Portfolio.py --cl")

    def test_runstring_cl_l(self):
        do_test_runstring("Portfolio.py --cl l")

    def test_runstring_cl_7(self):
        do_test_runstring("Portfolio.py --cl 7")

    def test_runstring_cl_debug_1(self):
        do_test_runstring("Portfolio.py --cl 7 --debug 1")
    '''


    # Testcases from docstring:


    def test_runstring_price_web_update(self):
        do_test_runstring("Portfolio.py --price_web_update --pfile demo_data_file.xml")

    def test_runstring_ac_web_update(self):
        do_test_runstring("Portfolio.py --ac_web_update --pfile demo_data_file.xml")

    def test_runstring_stats(self):
        do_test_runstring("Portfolio.py --stats --pfile demo_data_file.xml")

    def test_runstring_xml(self):
        do_test_runstring("Portfolio.py --xml --pfile demo_data_file.xml")

    def test_runstring_price_web_update_xml(self):
        do_test_runstring("Portfolio.py --price_web_update --xml --pfile demo_data_file.xml")

    def test_runstring_pt_show(self):
        do_test_runstring("Portfolio.py --pt show --pfile demo_data_file.xml")

    def test_runstring_pt_show_percent(self):
        do_test_runstring("Portfolio.py --pt show_percent --pfile demo_data_file.xml")

    def test_runstring_pt_show_ac_misc_totals(self):
        do_test_runstring("Portfolio.py --pt show_ac_misc_totals --pfile demo_data_file.xml")

    def test_runstring_pt_lp(self):
        do_test_runstring("Portfolio.py --pt lp --pfile demo_data_file.xml")

    def test_runstring_pt_add(self):
        do_test_runstring("Portfolio.py --pt add:'Small Value',Non_Retirement,50000:'Small Value',Retirement,20000:Healthcare,Retirement,50000 --pfile demo_data_file.xml")

    def test_runstring_inv_show(self):
        do_test_runstring("Portfolio.py --inv show --pfile demo_data_file.xml")

    def test_runstring_inv_diff_1file(self):
        do_test_runstring("Portfolio.py --inv diff --pfile demo_data_file.xml")

    def test_runstring_inv_diff_2file(self):
        do_test_runstring("Portfolio.py --inv diff --pfile demo_data_file.xml1,demo_data_file.xml2")

    def test_runstring_inv_lots(self):
        do_test_runstring("Portfolio.py --inv lots --pfile demo_data_file.xml")

    def test_runstring_all_not_silent(self):
        do_test_runstring("Portfolio.py --all --pfile demo_data_file.xml")

    def test_runstring_all_silent(self):
        do_test_runstring("Portfolio.py --all --silent --pfile demo_data_file.xml")

    def test_runstring_noindent(self):
        do_test_runstring("Portfolio.py --noindent  --all  --pfile demo_data_file.xml")

    def test_runstring_h(self):
        do_test_runstring("Portfolio.py --h")

    def test_runstring_xmlfile(self):
        do_test_runstring("Portfolio.py --xmlfile  --pfile demo_data_file.xml")

    def test_runstring_pdt(self):
        do_test_runstring("Portfolio.py --pdt --pfile demo_data_file.xml")

    def test_runstring_debug_1(self):
        do_test_runstring("Portfolio.py --debug 1  --all  --pfile demo_data_file.xml")

    def test_runstring_debug_2(self):
        do_test_runstring("Portfolio.py --debug 2  --all  --pfile demo_data_file.xml")

    def test_runstring_debug_12(self):
        do_test_runstring("Portfolio.py --debug 12  --all  --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show --pfile demo_data_file.xml")

    def test_runstring_price_web_update_inv_show(self):
        do_test_runstring("Portfolio.py --price_web_update --inv show --pfile demo_data_file.xml")

    def test_runstring_price_web_update_ac_web_update_pt_show(self):
        do_test_runstring("Portfolio.py --price_web_update --ac_web_update --pt show%lp%add:Small_Value,Retirement,13000%show_percent%lp_diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_lp(self):
        do_test_runstring("Portfolio.py --price_web_update --pt lp --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show_percent(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show_percent%lp%add:Large_Blend,Retirement,13000%show_percent%lp --pfile demo_data_file.xml")

    def test_runstring_price_web_update_inv_diff(self):
        do_test_runstring("Portfolio.py --price_web_update --inv diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show_percent_bigger_add(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show_percent%lp%add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000%show_percent%lp_diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_add(self):
        do_test_runstring("Portfolio.py --price_web_update --pt add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000%lp_diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show_ac_misc_totals(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show_ac_misc_totals --pfile demo_data_file.xml")






