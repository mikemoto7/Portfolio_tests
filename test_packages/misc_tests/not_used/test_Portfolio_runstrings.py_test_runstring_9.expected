
Processing: user_runstring = ./Portfolio.py --price_web_update --inv diff --pfile demo_data_file.xml

Processing: --price_web_update
NOTE: Detected in columnize_output(), row 2 is missing a column: 11 instead of 12.  Row: ['---Energy---', '', '', '', '', '', '', '', '', '', ''].  Will pad the row and continue.

Processing: --price_web_update --inv diff demo_data_file.xml 
---------------------------------------------------------------------------------------------------------------------------
Asset                           Symb  Own  iID  Account_ID  Ret_Class  Curr_Tot1  Curr_Tot2   ShPr1   ShPr2     Diff  Diff%
---Energy---                                                                                                               
Vanguard Energy                VGELX  os1   20       15777    Non_Ret   19898.00   19086.00   99.49   95.43   -812.0  -4.08
Vanguard Energy                VGENX  os1   42       35777   Trad_IRA   21208.00   20344.00   53.02   50.86   -864.0  -4.07
Vanguard Energy                VGENX  os1   52       45777   Roth_IRA   26510.00   25430.00   53.02   50.86  -1080.0  -4.07
Vanguard Energy                VGENX  os2   59       52777   Roth_IRA   30221.40   28990.20   53.02   50.86  -1231.2  -4.07
---Global---                                                                                                               
Fidelity Canada                FICDX  os2   39       34777   Trad_IRA   19995.30   19051.50   51.27   48.85   -943.8  -4.72
Vanguard Global Equity         VHGEX  os2    5        4777    SEP_IRA    2348.10    2520.90   26.09   28.01    172.8   7.36
Vanguard Global Equity         VHGEX  os2   23       18777    Non_Ret    6000.70    6442.30   26.09   28.01    441.6   7.36
Vanguard Global Equity         VHGEX  os2   47       40777   Trad_IRA   11740.50   12604.50   26.09   28.01    864.0   7.36
Vanguard Global Equity         VHGEX  os1   58       51777   Trad_IRA   14610.40   15685.60   26.09   28.01   1075.2   7.36
Vanguard Global Min Volatile   VMVFX  os1   24       19777    Non_Ret    2937.60    3086.40   12.24   12.86    148.8   5.07
---Healthcare---                                                                                                           
Vanguard Healthcare            VGHAX  os2    1         777    SEP_IRA    4135.50    4310.50   82.71   86.21    175.0   4.23
Vanguard Healthcare            VGHAX  os1   26       21777    Non_Ret   21504.60   22414.60   82.71   86.21    910.0   4.23
---Individual_Stocks---                                                                                                    
Netflix                         NFLX  os1   18       13777    Non_Ret   25878.60   28803.60  143.77  160.02   2925.0   11.3
---Large_Blend---                                                                                                          
Vanguard Dividend Growth       VDIGX  os2   45       38777   Trad_IRA   10367.30   10767.20   24.11   25.04    399.9   3.86
Vanguard Dividend Growth       VDIGX  os2   55       48777   Trad_IRA   12778.30   13271.20   24.11   25.04    492.9   3.86
Vanguard Total Stock Mkt Idx   VTSAX  os2   35       30777    Non_Ret   20363.00   21031.50   58.18   60.09    668.5   3.28
---Large_Growth---                                                                                                         
Fidelity Contrafund            FCNTX  os1   13        9777    Non_Ret   14644.00   15773.80   104.6  112.67   1129.8   7.72
Fidelity Contrafund            FCNTX  os1   36       31777   Trad_IRA   37656.00   40561.20   104.6  112.67   2905.2   7.72
Fidelity Contrafund            FCNTX  os1   38       33777  Trad_401K   39748.00   42814.60   104.6  112.67   3066.6   7.72
Vanguard Growth Idx            VIGAX  os2   25       20777    Non_Ret   15197.50   16262.50   60.79   65.05   1065.0   7.01
---Large_Value---                                                                                                          
Vanguard Equity Income         VEIRX  os2   21       16777    Non_Ret   14628.60   15012.90   69.66   71.49    384.3   2.63
Vanguard Equity Income         VEIPX  os1   46       39777   Trad_IRA   14621.20   15008.40   33.23   34.11    387.2   2.65
Vanguard Equity Income         VEIPX  os2   57       50777   Trad_IRA   18276.50   18760.50   33.23   34.11    484.0   2.65
---Mid_Blend_Aggress---                                                                                                    
Vanguard Extended Mkt Idx      VEXAX  os1   22       17777    Non_Ret   16662.80   16827.80   75.74   76.49    165.0   0.99
Vanguard Mid-Cap Idx           VIMAX  os2   27       22777    Non_Ret   46213.20   47341.80  171.16  175.34   1128.6   2.44
Vanguard Strategic Equity      VSEQX  os2   33       28777    Non_Ret   11097.90   11041.80   33.63   33.46    -56.1  -0.51
---Mid_Growth---                                                                                                           
Fidelity New Millennium        FMILX  os2   14       10777    Non_Ret    5613.00    5737.50   37.42   38.25    124.5   2.22
Vanguard Capital Opportunity   VHCOX  os2   19       14777    Non_Ret   10831.90   11538.70   57.01   60.73    706.8   6.53
Vanguard Capital Opportunity   VHCAX  os1   50       43777   Rollover   63168.00   67305.60   131.6  140.22   4137.6   6.55
Vanguard Mid-Cap Growth        VMGRX  os1    8        7777   Trad_IRA    2817.60    2931.60   23.48   24.43    114.0   4.05
Vanguard Mid-Cap Growth        VMGRX  os2    9        8777   Trad_IRA    3052.40    3175.90   23.48   24.43    123.5   4.05
---Mid_Value---                                                                                                            
Fidelity Low-Priced Stock      FLPSX  os2   37       32777   Trad_IRA   18825.60   19439.80   50.88   52.54    614.2   3.26
Vanguard Capital Value         VCVLX  os1   44       37777   Trad_IRA    5161.80    5216.40   12.29   12.42     54.6   1.06
Vanguard Capital Value         VCVLX  os1   54       47777   Trad_IRA    6390.80    6458.40   12.29   12.42     67.6   1.06
Vanguard Mid-Cap Value         VMVAX  os1   28       23777    Non_Ret   14677.60   14826.00   52.42   52.95    148.4   1.01
Vanguard Selected Value        VASVX  os1   30       25777    Non_Ret    9027.00    9198.00   30.09   30.66    171.0   1.89
---REIT---                                                                                                                 
Vanguard REIT                  VGSLX  os1    4        3777    SEP_IRA    9502.40    9244.80  118.78  115.56   -257.6  -2.71
Vanguard REIT                  VGSIX  os2    7        6777   Trad_IRA    3062.40    2978.80   27.84   27.08    -83.6  -2.73
Vanguard REIT                  VGSLX  os2   29       24777    Non_Ret   34446.20   33512.40  118.78  115.56   -933.8  -2.71
Vanguard REIT                  VGSIX  os2   43       36777   Trad_IRA   11414.40   11102.80   27.84   27.08   -311.6  -2.73
Vanguard REIT                  VGSLX  os2   49       42777   Trad_IRA   55826.60   54313.20  118.78  115.56  -1513.4  -2.71
---Small_Blend---                                                                                                          
Vanguard Small Idx             NAESX  os2    3        2777   Trad_IRA    4477.20    4466.70   63.96   63.81    -10.5  -0.23
Vanguard Tax-Managed Sml Val   VTMSX  os1   34       29777    Non_Ret   19019.60   18910.80   55.94   55.62   -108.8  -0.57
---Small_Growth---                                                                                                         
Vanguard Small Growth          VSGAX  os2   51       44777   Rollover   24093.30   24794.00   49.17    50.6    700.7   2.91
Vanguard Small Growth          VSGAX  os2   53       46777   Roth_IRA   25076.70   25806.00   49.17    50.6    729.3   2.91
Vanguard Small Growth Idx      VSGAX  os2   31       26777    Non_Ret   15242.70   15686.00   49.17    50.6    443.3   2.91
Vanguard Small Growth Idx      VISGX  os1   48       41777   Trad_IRA   18091.80   18616.20   39.33   40.47    524.4    2.9
---Small_Value---                                                                                                          
Vanguard Small-Cap Val Idx     VISVX  os1    6        5777   Trad_IRA    2970.00    2927.00    29.7   29.27    -43.0  -1.45
Vanguard Small-Cap Val Idx     VSIAX  os1   32       27777    Non_Ret   17036.80   16784.00   53.24   52.45   -252.8  -1.48
---Stats---                                                                                                                
Totals                            49                                    859068.8   878215.9                  19147.1       
Totals diff percent                                                                                                    2.23
Avg diff percent                                                                                                       2.27
Count pos diff percent                                                                                                   34
Count neg diff percent                                                                                                   15
Count neutral diff percent                                                                                                0

